### DISCUSSION TITLE
Get header from IOFactory and display it

### DISCUSSION THREAD

**Discussion by xavierpacotte** [dompdf/dompdf#2628] [Q&A]
I would like to know how to get the header of the file i'm using as IOFactory after saving it. I'm looking at the save function  to force it but I don't know how could I bring it to the DOM. I did a little test to display the page number as below and it works but it's page text. So I would like to do the same kind of thing but with the header section of $this

        //  Create PDF
        $pdf = $this->createExternalWriterInstance();
        $pdf->setPaper(strtolower($paperSize), $orientation);
        $pdf->loadHtml(str_replace(PHP_EOL, '', $this->getContent()));
        $pdf->render();

       //test
        $canvas = $pdf->get_canvas();
        $canvas->page_text(72, 18, "Header: {".PAGE_NUM."} of {PAGE_COUNT}",
                        "helvetica", 14, array(0,0,0));
        
        //  Write to file
        fwrite($fileHandle, $pdf->output());
        parent::restoreStateAfterSave($fileHandle);

**Comment by bsweeney** [comment]
This doesn't sound like a Dompdf question. Can you provide more detail on what you're trying to achieve and what framework you're using?

**Reply by xavierpacotte** [reply]
I'm trying to get the header and footer from a template using IOFactory::load() but when I save the file as PDF or HTML i don't see them and if I use `dd()` I can see them in a section, the problem is that the function `getContent() `only look for the elements and not the header and footer I think. Is there any way to display them ?

**Reply by bsweeney** [reply]
Can you provide more context? Dompdf doesn't use a class called IOFactory.

If you're talking about [PHPOffice](https://github.com/PHPOffice/) you may want to ask there. Or check out StackOverflow for answers.
